services/config_data_sources/json_project_data_source.rb

```diff
@@ -158,14 +158,14 @@ def save_git_repo_configs!(git_repo_configs: nil)
     end
 
     def create_project!(name: nil, repo_config: nil, enabled: nil, lane: nil, artifact_provider: nil)
-      projects = self.projects
+      projects = self.projects.clone
```

Comment:
I think this will break, `self.projects` is an array, right?

Reply:
As far as I've tested it, doesn't seem to break, `.dup` just makes a light copy, but `.clone` makes a deep copy of the whole object (and its contents). That being said, just as far as I've tested so far. If we are not sure this can break at some moment let's just revert it 😄

Reply:
Ohhh, no no, I was confused. I was thinking you wanted to call `.clone` on a git repo, not the actual Ruby `Object.clone`. That being said, it looks like this is just a shallow clone too 😆 

```
Produces a shallow copy of obj—the instance variables of obj are copied, but not the objects they reference. clone copies the frozen 
```
(https://ruby-doc.org/core-2.3.3/Object.html#method-i-clone)